source/Halibut.Tests/Transport/Streams/ErrorRecordingStreamFixture.cs

```diff
@@ -113,6 +113,22 @@ public async Task EndOfStreamEncounteredIsRecordedAsync()
             errorRecordingStream.ReadExceptions.Count.Should().Be(0);
         }
         
+        [Test]
+        public async Task ReadingIntoAZeroLengthArrayIsNotAEndOfStream()
```

Comment:
This test seems to be testing a few test cases:
- 100 length array with 0 count
- 0 length array (which has to have a 0 count)

Maybe we should split these test cases up?

Also, should we also verify the return value to be 0 as well?